src/shake_alert/shake_alert.go

```diff
@@ -0,0 +1,104 @@
+package shake_alert
+
+import (
+	"math"
+	"sort"
+)
+
+// constants
+const (
+	windowSize = 391
+	k          = 5 //# of neighbors
+	//r          = 15000
+)
+
+// finds p-th percentile of data
+func findPercentile(data []float64, p float64) float64 {
+	if len(data) == 0 {
```

Comment:
Similar to previous cases, is it okay if this function sorts the input? Another thing to consider is that if you inadvertently sort the input data multiple times when you use a combination of these functions in your code.

Reply:
like with `findMean()`, if there are no values in the array then it might not make sense to return a usable value; NaN may make more sense: https://go.dev/play/p/TnJpgoXNZHp